### DISCUSSION TITLE
Also using win32API, why does globalmousekeyhook doesn't work in the game, but this project does!

### DISCUSSION THREAD

**Discussion by sanshi2018** [Jagailo/YetAnotherKeyDisplayer#26] [Q&A] [answered]
I've been looking for a program to capture keyboard and mouse in games. I found now [this ](https://github.com/Jagailo/YetAnotherKeyDisplayer)program and [globalmousekeyhook](https://github.com/gmamaladze/globalmousekeyhook), [SharpDX.RawInput](https://github.com/sharpdx/SharpDX-Samples/blob/master/Desktop/RawInput/RawInputTrackApp/Program.cs).
  globalmousekeyhook and this project have similar technical solutions, both use SetWindowsHookEx . But I can't figure out why this project works fine in the game but globalmousekeyhook doesn't!

Your core code is

```
private void Install()
{
    if (_hookHandle != IntPtr.Zero)
    {
        return;
    }
    var modules = Assembly.GetExecutingAssembly().GetModules();
    System.Diagnostics.Debug.Assert(modules != null && modules.Length > 0);

    _hookHandle = SetWindowsHookEx(HookType, _hookFunction, Marshal.GetHINSTANCE(modules[0]), 0);
}
```

globalmousekeyhook core code is

```

private static HookResult HookGlobal(int hookId, Callback callback)
{
    _globalHookProc = (code, param, lParam) => HookProcedure(code, param, lParam, callback);

    var hookHandle = HookNativeMethods.SetWindowsHookEx(
        hookId,
        _globalHookProc,
        Process.GetCurrentProcess().MainModule.BaseAddress,
        0);

    if (hookHandle.IsInvalid)
        ThrowLastUnmanagedErrorAsException();

    return new HookResult(hookHandle, _globalHookProc);
}

```

**Comment by sanshi2018** [comment]
This project is really awesome! ! ! Thanks! I found a lot of projects that don't work in the game. this is the only one

**Comment by sanshi2018** [comment, marked as answer]
I think I know why, in fact both globalmousekeyhook and SharpDX.RawInput work fine in the game, with the prerequisite requirement that it needs to be run as administrator. I'm really laughing at my own stupidity for being tossed around for so long by a minor issue

**Comment by Jagailo** [comment]
Glad to hear that 🙂
